File: python/meteroid/models/plan_add_on_input.py
# this file is @generated
import dataclasses
import typing as t

from ..serialization import BaseModel
from .add_on_id import AddOnId
from .price_id import PriceId


@dataclasses.dataclass
class PlanAddOnInput(BaseModel):
    add_on_id: AddOnId

    max_instances: t.Optional[int] = None

    price_id: t.Optional[PriceId] = None

    self_serviceable: t.Optional[bool] = None

File: go/plan_add_on_input.go
// this file is @generated
package meteroid

type PlanAddOnInput struct {
	AddOnId AddOnId `json:"add_on_id"`

	MaxInstances *int32 `json:"max_instances,omitempty"`

	PriceId *PriceId `json:"price_id,omitempty"`

	SelfServiceable *bool `json:"self_serviceable,omitempty"`
}
